File: ios/SwitchViewManager.m
//
//  SwitchViewManager.m
//  rn_native_modules
//
//  Created by Hihi on 02/12/2022.
//

#import <Foundation/Foundation.h>
#import "React/RCTViewManager.h"

@interface RCT_EXTERN_MODULE(SwitchViewManager, RCTViewManager)

RCT_EXPORT_VIEW_PROPERTY(label, NSString)
RCT_EXPORT_VIEW_PROPERTY(isOn, CFBoolean)
RCT_EXPORT_VIEW_PROPERTY(onValueChange, RCTBubblingEventBlock)

@end

File: ios/SwitchViewManager.swift
//
//  SwitchViewController.swift
//  rn_native_modules
//
//  Created by Hihi on 01/12/2022.
//
import UIKit

@objc(SwitchViewManager)
class SwitchViewManager : RCTViewManager {
    
  override func view() -> UIView! {
    return SwitchView()
  }
  
  override static func requiresMainQueueSetup() -> Bool {
      return true
  }
}
